base/parse.jl

```diff
@@ -167,6 +167,32 @@ float(x::AbstractString) = parse(Float64,x)
 
 float{S<:AbstractString}(a::AbstractArray{S}) = map!(float, similar(a,typeof(float(0))), a)
 
+function parse{T<:AbstractFloat}(::Type{T}, s::AbstractString, base::Integer)
+    s = strip(s, ' ')
```

Comment:
I think the function should strip. And if you reuse the `Integer` case there will be no allocation.